### PR TITLE
CLN: Removed second version of xstrtod in parser_helper.h

### LINKED ISSUE #19361: Duplicate xstrtod implementations
In _libs/src/parse_helper.h and libs/src/parser/tokenizer.c

Not identical, comments suggest the tokenizer.c version may be more modern.

### PR DESCRIPTION
**Pull request by anmyachev**
- [x] closes #19361
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by gfyoung** [comment]
@anmyachev : Thanks for the PR!  Do mind running `asv` to see if performance is impacted with this de-duplication?

**Comment by jbrockmendel** [comment]
AFAICT this looks fine, but I'm going to kick the review over to @chris-b1 who is better-versed in this area.

**Comment by gfyoung** [comment]
@jbrockmendel: Sounds good.  Was just pinging for the modified dependency linking (given your focus on reorganizing internal dependencies to make it more DAG)

**Comment by anmyachev** [comment]
@gfyoung after this run:
`asv continuous -f 1.01 origin/master HEAD -e -b inference.MaybeConvertNumeric ` 
result: `BENCHMARKS NOT SIGNIFICANTLY CHANGED.`
Same result for run `asv continuous -f 1.01 origin/master HEAD -e -b io.csv`
- First run check perf of xstrtod with `maybe_int != NULL` 
- Second run check perf of xstrtod with `maybe_int == NULL`

File: pandas/_libs/src/parser/tokenizer.c [hunk 1]
```diff
@@ -1554,6 +1554,7 @@ double xstrtod(const char *str, char **endptr, char decimal, char sci,
     int n;
     int num_digits;
     int num_decimals;
+    int _maybe_int = 1;
```
Comment (jreback) [inline pandas/_libs/src/parser/tokenizer.c]: I find this kind of indirection extra confusing, pls don't do this.

File: pandas/_libs/src/parser/tokenizer.c [hunk 2]
```diff
@@ -1688,6 +1693,7 @@ double precise_xstrtod(const char *str, char **endptr, char decimal, char sci,
     int num_decimals;
     int max_digits = 17;
     int n;
+    int _maybe_int = 1;
```
Comment (jreback) [inline pandas/_libs/src/parser/tokenizer.c]: same

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/_libs/src/parser/tokenizer.c (hunk 1 above)
Comment (anmyachev) [reply pandas/_libs/src/parser/tokenizer.c]: removed extra _maybe_int variable

File: pandas/_libs/src/parser/tokenizer.c (hunk 2 above)
Comment (anmyachev) [reply pandas/_libs/src/parser/tokenizer.c]: also removed

**Comment by jreback** [comment]
@chris-b1 any comments here

**Review by chris-b1 — APPROVED** [review]
lgtm!

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @anmyachev

**Pull request merged**